Cookbook/Proposals/Comments.md

```diff
@@ -0,0 +1,224 @@
+    public enum State: String, Codable {
+        /// waiting for payment
+        case pending
+        /// user failed to pay for appoitment withing 10 min after the booking
+        case timedOut = "timed_out"
+        /// canceled by user or doctor
+        case cancelled
+        /// paid by user
+        case paid
+        /// doctor finished adding notes on the portal
+        case completed
+        /// patient did not answer the doctor's call
+        case noShow = "no_show"
+    }
+}
+```
+Can be simplified without any lose in understanding what the code represents to:
+
+```swift
+extension AppointmentDTO {
+    public enum State: String, Codable {
+        /// waiting for payment
+        case waitingForPayment = "pending"
+        /// timeout == 10 min
+        case paymentTimedOut = "timed_out"
+        case cancelledByUserOrDoctor = "cancelled"
+        case paidByUser = "paid"
+        case doctorFinishedAddingNotesOnThePortal = "completed"
+        case patientDidntAnswerTheCall = "no_show"
+    }
+}
+```
+
+## Proposed solution
+
+### Our approach 
+The most important thing is to improve the definition of our approach to comments as currently it isn't said clearly. Moreover, it would be nice to have a common way how we comment our code when we are forced to add them. 
+
+In my opinion, our approach should insist on writing always a **self-documented code**. It should be our goal. However, writing self-documented code **doesn't** mean we shouldn't use comments. It means, that commenting our code is a **necessary evil**:
+
+> The proper use of comments is to compensate for our failure to express ourself in code. Note that I used the word failure. I meant it. Comments are always failures. We must have them because we cannot always figure out how to express ourselves without them, but their use is not a cause for celebration.
+    // Robert C. Martin - "Clean Code"
+    
+I think that adding the above section to our style guide would make it clear what's our approach to comments.
+
+### What is a good comment?
+
+To make our StyleGuide more verbose we can also add a list of what kind of comments we think are "good".
+
+**Good comments**:
+
+- **TODO comments**
+    This already has a good definition of our approach in the StyleGuide and I think we are fine with that approach.
+- **Doc headers for public API**
+    Good documentation helps consumers of the SDK to understand what the code does. However, I don't know if it's worth to add a comment to every public function/field, even if a comment is just a repetition. The following comment doesn't add any value (unless it's needed by a doc generation tool if we use any):
+    ```swift
+     /// Supported biometric types
+     public var supportedBiometry: BiometryType
+    ```
+- **Explanation of Intent/Clarification**
+    It's a kind of comment when a developer tries to explain what he's trying to achieve when a code is not verbose enough. Example: Math code, sometimes parsing.
```

Comment:
I feel like those kind of comments are the main ones we overuse, explaining the code while trying to make the code self-explanatory should be the better approach. So not sure I'd list those in the "Good comments" category in that regard...

```suggestion
    It's a kind of comment when a developer tries to explain what they're trying to achieve when a code is not verbose enough. Example: Math code, sometimes parsing.
```

Reply:
@AliSoftware I agree! However, I think such a comment is better than not adding it and not providing good naming. The comment is fine however,  we still can disagree with putting it there. That's why a few lines below I've written :

> The above list is what defines a good comment. **However**, we still have to remember that **every** comment **is our failure** to name our code better. If someone gives you a good hint how to remove a comment without sacrificing readability **we should go for it**.